src/Raven.Server/Documents/CountersStorage.cs

```diff
@@ -317,11 +318,23 @@ public string IncrementCounter(DocumentsOperationContext context, string documen
             
             using (GetCounterKey(context, documentId, name, context.Environment.Base64Id, out var counterKey))
             {
-                long prev = 0;
+                var value = delta;
                 if (table.ReadByKey(counterKey, out var existing))
                 {
-                    prev = *(long*)existing.Read((int)CountersTable.Value, out var size);
+                    var prev = *(long*)existing.Read((int)CountersTable.Value, out var size);
                     Debug.Assert(size == sizeof(long));
+                    try
+                    {
+                        value = checked(prev + delta); //inc
+
+                    }
+                    catch (OverflowException e)
+                    {
+                        throw new CounterOverflowException(
+                            $"Could not increment counter '{name}' from document '{documentId}' " +
+                            $"with value '{prev}' by '{delta}'. " +
+                            "Arithmetic operation resulted in an overflow.", e);
```

Comment:
The last sentence should be removed, that is the message of the overflow exception.

Reply:
Fixed